src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -4096,7 +4096,7 @@ void InstanceKlass::set_init_state(ClassState state) {
   assert(good_state || state == allocated, "illegal state transition");
 #endif
   assert(_init_thread == nullptr, "should be cleared before state change");
-  _init_state = state;
+  Atomic::release_store_fence(&_init_state, state);
```

Comment:
Why not just a release_store ?? Why do we need the trailing fence?

Reply:
Says in PR: "Current patch makes a seqcst write, which is stronger than strictly necessary. I think it is okay to be extra paranoid on rarely-executed class initialization path." But I can turn it into just a weaker release, sure.

Reply:
I thought a seqcst write would be `fence(); store; fence()`?

Anyway I don't like "paranoid when it comes to memory barriers because that says to me "hey we don't understand what is going on here so we're just going to do the heaviest barrier we can 'just in case'."